src/middlewared/middlewared/plugins/filesystem_/acl_base.py

```diff
@@ -0,0 +1,409 @@
+        expectations regarding permissions inheritance. This entry is removed from NT ACL returned
+        to SMB clients when 'ixnas' samba VFS module is enabled. We also remove it here to avoid confusion.
+        """
+
+    @accepts(
+        Dict(
+            'filesystem_ownership',
+            Str('path', required=True),
+            Int('uid', null=True, default=None),
+            Int('gid', null=True, default=None),
+            Dict(
+                'options',
+                Bool('recursive', default=False),
+                Bool('traverse', default=False)
+            )
+        )
+    )
+    @job(lock="perm_change")
+    def chown(self, job, data):
+        """
+        Change owner or group of file at `path`.
+
+        `uid` and `gid` specify new owner of the file. If either
+        key is absent or None, then existing value on the file is not
+        changed.
+
+        `recursive` performs action recursively, but does
+        not traverse filesystem mount points.
+
+        If `traverse` and `recursive` are specified, then the chown
+        operation will traverse filesystem mount points.
+        """
+
+    @accepts(
+        Dict(
+            'filesystem_permission',
+            Str('path', required=True),
+            UnixPerm('mode', null=True),
+            Int('uid', null=True, default=None),
+            Int('gid', null=True, default=None),
+            Dict(
+                'options',
+                Bool('stripacl', default=False),
+                Bool('recursive', default=False),
+                Bool('traverse', default=False),
+            )
+        )
+    )
+    @job(lock="perm_change")
+    def setperm(self, job, data):
+        """
+        Remove extended ACL from specified path.
+
+        If `mode` is specified then the mode will be applied to the
+        path and files and subdirectories depending on which `options` are
+        selected. Mode should be formatted as string representation of octal
+        permissions bits.
+
+        `uid` the desired UID of the file user. If set to None (the default), then user is not changed.
+
+        `gid` the desired GID of the file group. If set to None (the default), then group is not changed.
+
+        `stripacl` setperm will fail if an extended ACL is present on `path`,
+        unless `stripacl` is set to True.
+
+        `recursive` remove ACLs recursively, but do not traverse dataset
+        boundaries.
+
+        `traverse` remove ACLs from child datasets.
+
+        If no `mode` is set, and `stripacl` is True, then non-trivial ACLs
+        will be converted to trivial ACLs. An ACL is trivial if it can be
+        expressed as a file mode without losing any access rules.
+
+        """
+
+    @accepts()
+    async def default_acl_choices(self):
+        """
+        Get list of default ACL types.
+        """
+
+    @accepts(
+        Str('acl_type', default='NFS4_OPEN' if osc.IS_FREEBSD else 'POSIX_OPEN', enum=[x.name for x in ACLDefault]),
```

Comment:
This changes the API in a non-backward compatible way.

Thinking if we could make alias for the previous values matching NFS4. Thoughts?

Reply:
Sounds like a good idea. I'll implement in morning.